newton/_src/utils/download_assets.py

```
@@ -82,16 +112,35 @@ def download_git_folder(
+
+    # 1. Handle force_refresh
+    if force_refresh and cache_folder.exists():
+        _safe_rmtree(cache_folder)
+
+    # 2. Check cache validity using Git
+    is_cached = target_folder.exists() and (cache_folder / ".git").exists()
+    if is_cached:
+        try:
+            repo = git.Repo(cache_folder)
+            current_commit = repo.head.commit.hexsha
+        except Exception:
```

Ruff (0.14.3)

Do not catch blind exception: `Exception`

(BLE001)